src/utils.ts

```diff
@@ -19,21 +19,29 @@ export const inIframe = (): boolean => {
   }
 };
 
+// FIXME: Not sure why custom type guards are used to detect v2 and v3 documents
 export const isV2Document = (document: any): document is v2.OpenAttestationDocument => {
   return !!document.$template;
 };
 
+// FIXME: Not sure why custom type guards are used to detect v2 and v3 documents
 export const isV3Document = (document: any): document is v3.OpenAttestationDocument => {
   return !!document["@context"];
 };
 
+export const isV4Document = (document: unknown): document is v4.OpenAttestationDocument =>
+  utils.isWrappedV4Document(document) || utils.isSignedWrappedV4Document(document);
+
 const getTemplateName = (document: OpenAttestationDocument): string => {
   if (isV2Document(document) && typeof document.$template === "object") {
     return document.$template.name;
   }
   if (isV3Document(document) && document.openAttestationMetadata.template) {
```

Comment:
I think we need to do a safe call here or a typeof check like for v2 - i'm getting undefined error for v4 docs (trying to read `.template` but is undefined)

Reply:
In OA v4, the renderer template object is now under `.renderMethod`. Could you check if you are able to obtain this object?

```jsonc
// Example:
{
  // ...
  "renderMethod": {
    "type": "OpenAttestationRenderMethod",
    "renderMethodType": "EMBEDDED_RENDERER",
    "name": "GOVTECH_DEMO",
    "url": "https://demo-renderer.openattestation.com"
  },
}
```

Reply:
i'm able to, but it is throwing an exception at this line before evaluating the `isV4Document...`

Reply:
Yeah I think the `isV3Document()` type guard is too loose and is wrongly categorising v4 as v3 documents.

Perhaps we can add an optional chaining `?` since `openAttestationMetadata` has been omitted in v4 already:

```suggestion
  if (isV3Document(document) && document.openAttestationMetadata?.template) {
```

Reply:
yup this is working for me

Reply:
Alternatively, I can modify the `isV3Document()` type guard to this:

```typescript
export const isV3Document = (document: any): document is v3.OpenAttestationDocument => {
  return !!document["@context"] && !!document["openAttestationMetadata"];
};
```